python/cudf/cudf/core/dataframe.py

```diff
@@ -1700,10 +1697,32 @@ def _concat(cls, objs, axis=0, ignore_index=False):
             )
         else:
             index = Index._concat([o.index for o in objs])
-        data = [
-            (c, Series._concat([o[c] for o in objs], index=index))
-            for c in objs[0].columns
-        ]
+
+        # Currently we only support sort = False
+        # Change below when we want to support sort = True
+        # below functions as an ordered set
+
+        all_columns_list = []
+        all_columns_set = set()
+        for o in objs:
+            for col in o.columns:
+                if col not in all_columns_set:
+                    all_columns_list.append(col)
+                    all_columns_set.add(col)
+
+        # Concatenate cudf.series for all columns
+        data = []
+
+        for c in all_columns_list:
+            series_list = [
+                o[c]
+                if c in o.columns
+                else utils.get_null_series(size=len(o), dtype=np.bool)
+                for o in objs
+            ]
+            concatenated_series = Series._concat(series_list, index=index)
+            data.append((c, concatenated_series))
```

Comment:
More list comprehensions:
```
data = [
    (
        c,
        Series._concat(
            [
                o[c]
                if c in o.columns
                else utils.get_null_series(size=len(o), dtype=np.bool)
                for o in objs
            ],
            index=index,
        ),
    )
    for c in all_columns_list
]
```

Reply:
Thanks for the review, will update this and push the required change.